api/src/main/java/org/ccci/gto/globalreg/BaseGlobalRegistryClient.java

```diff
@@ -112,7 +113,8 @@ private void attachFilters(final Request request, final Filter... filters) {
     }
 
     @Override
-    public final <T> T addEntity(@Nonnull final Type<T> type, @Nonnull final T entity, final Set<String> fields) throws GlobalRegistryException {
+    public final <T> T addEntity(@Nonnull final Type<T> type, @Nonnull final T entity, final Set<String> fields,
+                                 boolean requireMdm) throws GlobalRegistryException {
```

Comment:
I really don't like needing all these method variations with different params... But this PR is probably not the right place to address that ;)

Long term, I'm curious if Kotlin's support for Default and Named arguments could simplify this. https://kotlinlang.org/docs/reference/functions.html#default-arguments

Reply:
I certainly don't like it either.

The long-term solution in my mind is a fluent api.

Reply:
a fluent api would be a good solution too